_pytest/config.py

```diff
@@ -972,11 +972,31 @@ def _initini(self, args):
         ns, unknown_args = self._parser.parse_known_and_unknown_args(args, namespace=self.option.copy())
         r = determine_setup(ns.inifilename, ns.file_or_dir + unknown_args, warnfunc=self.warn)
         self.rootdir, self.inifile, self.inicfg = r
-        self._parser.extra_info['rootdir'] = self.rootdir
-        self._parser.extra_info['inifile'] = self.inifile
-        self.invocation_dir = py.path.local()
         self._parser.addini('addopts', 'extra command line options', 'args')
         self._parser.addini('minversion', 'minimally required pytest version')
+        self._parser.addini('pathtype', 'path implementation to be used', default='pylib')
+
+        self.invocation_dir = self.make_path()
+        self.rootdir = self.make_path(self.rootdir)
```

Comment:
> this is why this will be and opt-in and will support a transitional api for quite a while
> this is something that will have a transition period way beyond the lifetime of python 2

Oh OK that seems sensible, and as soon as we add an alternative the better. But what do you mean by a "transitional API" exactly? What you did here or something else entirely like a `py.local  subclass which supports `pathlib.Path` methods?

> however py.path has to die

Yeah, it had its purpose of course but with `pathlib` on standard library it doesn't make sense to maintain it in the long run.